manticore/platforms/linux.py

```diff
@@ -2408,6 +2441,80 @@ def sys_open(self, buf, flags, mode):
 
         return rv
 
+    def sys_openat(self, dirfd, buf, flags, mode):
+        '''
+        A version of openat that includes a symbolic path and symnbolic directory file descriptor
+
+        :param dirfd: directory file descriptor
+        :param buf: address of zero-terminated pathname
+        :param flags: file access bits
+        :param mode: file permission mode
+        '''
+
+        if issymbolic(dirfd):
+            logger.debug("Ask to read from a symbolic directory file descriptor!!")
+            raise ConcretizeArgument(self, 0)
+
+        if issymbolic(buf):
+            logger.debug("Ask to read to a symbolic buffer")
+            raise ConcretizeArgument(self, 1)
+
+        return super(SLinux, self).sys_openat(dirfd, buf, flags, mode)
+
+    def sys_getrandom(self, buf, size, flags):
+        '''
+        The getrandom system call fills the buffer with random bytes of buflen.
+        The source of random (/dev/random or /dev/urandom) is decided based on the flags value.
+
+        :param buf: buffer to be filled with random bytes
+        :param size: size of the random bytes to be filled
+        :param flags: source of random (/dev/random or /dev/urandom)
+        :return: number of bytes copied to buf
+        '''
+        data = ''
+        if issymbolic(buf):
+            logger.debug("Ask to generate random to a symbolic buffer")
+            raise ConcretizeArgument(self, 1)
+
+        if issymbolic(size):
+            logger.debug("Ask to generate random of symbolic number of bytes ")
+            raise ConcretizeArgument(self, 2)
+
+        try:
+          flag = ['', 'GRND_NONBLOCK', 'GRND_RANDOM'][flags & 7]
```

Comment:
Assign names to `flag` values and use those in the body, as opposed to turning them into a string and turning flag checking into a string comparison? Also, the way it's written right now, the code is incorrect in case multiple flags are set. (line 2499 is never reached)

i.e.
```
flags = flags & 7
GRND_NONBLOCK = 1
GRND_NONBLOCK = 2
...
if flags & GRND_NONBLOCK:
    ..
```

Reply:
Will rework the flags logic.

Reply:
Any updates on the above rework? We can also help with implementing it.